### DISCUSSION TITLE
Nikon NEF to Tiff

### DISCUSSION THREAD

**Discussion by brettdeegarrett** [dlemstra/Magick.NET#1421] [Help]
I have a NEF file that I open in Nikon NX Studio and export using defaults as an 8 bit tiff
I am attempting to use Magick.NET to take the same NEF file and get a very close result as NX Studio exports. The image I get from Magic.NET is more yellowish and the black lettering of the words are not as black as  what NX Studio exported tiff image. 
I have tried a lot of different things based on searching the existing Discussions. 
My Code:
```C#
public void ConvertNEFtoTIFFMagick(string inputFilePath, string outputFilePath)
	{
		MagickReadSettings readSettings = new MagickReadSettings
		{
			Format = MagickFormat.Nef,
			//ColorSpace = ColorSpace.sRGB
			//ColorSpace = ColorSpace.RGB
		};
		readSettings.SetDefine(MagickFormat.Nef, @"use-camera-wb", true); //true lightened up the image a bit
		readSettings.SetDefine(MagickFormat.Nef, @"no-auto-bright", false); //true darkened the background
		//readSettings.SetDefine(MagickFormat.Nef, @"use-auto-wb", true); //Not much effect true or false
		readSettings.SetDefine(MagickFormat.Nef, @"output-color", 2); // 0 Raw,1 SRGB, 2 Adobe RGB, 3 WideGamutRGB, 4 KodakProPhotoRBG
       // readSettings.SetDefine(MagickFormat.Nef, @"interpolation-quality",3); //-1 disabled 0 Linear, 1 Vng, 2 Ppg, 3 Ahd, 4 Dcb
		//readSettings.SyncImageWithTiffProperties= true;
		using (MagickImage image = new MagickImage(inputFilePath, readSettings))
		{
			// Set the output format to TIFF
			image.Format = MagickFormat.Tiff;
			//image.ColorSpace = ColorSpace.sRGB;
			image.Depth = 8;

			// Save the image as a TIFF file
			image.Write(outputFilePath);
		}
	}
```

Note: The image is captured using a "Nikon Custom" Picture Control. I have access to the c# object that was set the "Picture Control" on the camera and it includes a custom curve. I am trying to figure out if I need to somehow use this information to get the result I want.
The following three files can be found on google drive at the following URL  https://drive.google.com/drive/folders/1jKECcxhOYQVTjl2jYDy5eVpPxnwwc97U?usp=sharing
1. Original NEF  testRaw.nef
2. NX Studio converted testRaw_NXStudio.tiff
3. testRawAllMagick.tiff

**Comment by dlemstra** [comment]
Those calls to `SetDefine` should not do anything. Have you tried to use `DngReadDefines` instead?
